artifact.md

```diff
@@ -0,0 +1,84 @@
+# OCI Artifact Manifest Specification
+
+The goal of the Artifact Manifest Specification is to define content addressable artifacts in order to store them along side container images in a registry. Like [OCI Images](manifest.md), OCI Artifacts may be referenced by the hash of their manifest. Unlike OCI Images, OCI Artifacts are not meant to be used by any container runtime.
+
+Examples of artifacts that may be stored along with container images are Software Bill of Materials (SBOM), Digital Signatures, Provenance data, Supply Chain Attestations, scan results, and Helm charts.
+
+This section defines the `application/vnd.oci.artifact.manifest.v1+json` [media type](media-types.md).
+For the media type(s) that this is compatible with see the [matrix](media-types.md#compatibility-matrix).
+
+# Artifact Manifest
+
+## *Artifact Manifest* Property Descriptions
+
+- **`mediaType`** *string*
+
+  This property MUST be used and contain the media type `application/vnd.oci.artifact.manifest.v1+json`.
+
+- **`artifactType`** *string*
+
+  This property SHOULD specify the mediaType of the referenced content and MAY be registered with [IANA][iana].
+
+- **`blobs`** *string*
+
+  This OPTIONAL property contains a list of [descriptors](descriptor.md).
+  Each descriptor represents an artifact of any IANA mediaType.
+  The list MAY be ordered for certain artifact types like scan results.
+
+- **`refers`** *string*
+
+  This OPTIONAL property specifies a [descriptor](descriptor.md) of a container image or another artifact.
+  The purpose of this property is to provide a reference to the container image or artifact this artifact is related to.
+  The "Referrers" API in the distribution specification looks for this property to list all artifacts that refer to a given artifact or container image.
+
+- **`annotations`** *string-string map*
+
+  This OPTIONAL property contains additional metadata for the artifact manifest.
+  This OPTIONAL property MUST use the [annotation rules](annotations.md#rules).
+
+  The following annotations MAY be used:
+
+  - `org.opencontainers.artifact.description`: human readable description for the artifact
+  - `org.opencontainers.artifact.created`: creation time of the artifact expressed as string defined by [RFC 3339][rfc-3339]
+
+  Additionally, the following annotations SHOULD be used when deploying multi-arch container images:
+
+  - `org.opencontainers.platform.architecture`: CPU architecture for binaries
+  - `org.opencontainers.platform.os`: operating system for binaries
+  - `org.opencontainers.platform.os.version`: operating system version for binaries
+  - `org.opencontainers.platform.variant`: variant of the CPU architecture for binaries
+
+    Also, see [Pre-Defined Annotation Keys](annotations.md#pre-defined-annotation-keys).
+
+  User defined annotations MAY be used to filter various artifact types, e.g. signature public key hash, attestation type, and SBOM schema.
```

Comment:
As the spec allows user defined annotations, do we want to cap the number of annotations stored or returned as part of `referrers` API ?

Reply:
The current proposal is that all annotations would be returned. If we were to trim down the list, then we need define the criteria.  For example, only the first 10 annotations would be returned.  That still doesn't cap the size of the payload since a single key could actually be large enough. 

I think it would be good to raise this issue in the distribution spec since the referrers response payload and registry implementations needs to now ensure that data from all manifests are returned.  https://github.com/opencontainers/distribution-spec/pull/335 /cc @michaelb990

Reply:
The index we return is rather minimal, so having enough annotations to cause this to exceed the recommended 4MB limit is likely to also exceed that limit in the referrer's manifest too. I think any limits we impose should be set on the original manifest, and not this API, to minimize complexity.

Reply:
Lets say worst case, there are like 20 artifacts in the referrers API reponse and each had say about ~3MB of annotations, then the API reponse payload would be very large. In these cases what should the API return? May be capping the reponse size per artifact?

Reply:
At that point, the response would trigger pagination, with one descriptor per page. It also wouldn't be supported to include more than one of those descriptors in the tag schema fallback. My hope is that manifests like that result in a bad enough UX, that bug reports get opened on the source and alternatives are used.

The downside of capping what is returned in the API response, is that any partial descriptor would then need to be pulled as a full manifest to determine if the client side filters match (the filtered annotation could be one that was trimmed). We would also need to provide an indicate that the descriptor is a partial response so clients know when they need to fall back to pulling the full manifest directly. When thinking of client implementations, that would probably be one pull for the API, one of the manifest by the artifact listing method, and then another pull of the manifest by the artifact get method, unless the manifest is cached between those methods.